### ISSUE TITLE
high-frequency has machine noise

### ISSUE THREAD

**Issue by hdmjdp** [jik876/hifi-gan#16]
![图片](https://user-images.githubusercontent.com/26831425/99934861-33128f80-2d9a-11eb-8d0d-bddce32e0f79.png)

hi, this a 48k wav trained 120k steps, you can see the noise above 16kHz.   How can I remove this? thanks

**Comment by jik876** [comment]
Hi, I think 120k steps is too early for the model to synthesize high quality audio.
And it would be nice to make sure that fmax isn't low, because low fmax will lead to create partially obscured input conditions.

**Comment by hdmjdp** [comment]
> fmax

fmax=null，so default famx will be 48000/2=24000?
librosa.filters.mel(sample_rate, n_fft, n_mels=num_mels, fmin=fmin)

**Comment by jik876** [comment]
@hdmjdp 

In our experiments, the model was able to synthesize fairly good quality audio at 500k steps. It depends on the dataset and experiment settings, but it would be a good idea to see if the noise is still high at similar training steps.
And It would also be a good idea to experiment with more than 80 bands, because 48kHz - 80bands spectrogram can be seen as less detailed than 22kHz - 80bands spectrogram.

**Comment by hdmjdp** [comment]
@jik876 Yes，actually, I have used 128 bands. And whether more DiscriminatorP can help to improve the quanlity of 48k audio?

**Comment by jik876** [comment]
@hdmjdp 

In general, more period discriminators with different periods would be helpful to improve quality, but there are trade-offs such as training speed.
In your case, I would recommend training the model sufficiently first.
I hope you have good results.

**Comment by DeepDubbed** [comment]
My finding is that if your input is an 80 bands,  fmax 8000 hz mel-spectogram, you will not be able to generate as good quality as possible of 22khz audio.  That's because the network doesn't see any frequencies above 8000 hz and you will miss about 3025 hz that the network has to "invent". I don't think this generator is good for "super-resolution"
 
80 bands, fmax 8000 hz mel-spectrograms are good for 16khz sample-rates - and it "virtually" maps every 100 Hz to one band (it's a bit more complex due to mel-scale, triangulation, etc etc.). The reason why it might had been decided to use 80 bands, fmax 8000 hz for inputs is probably to match synths that had been trained / designed to work on 16 khz sample-rates and that generate that kind of output. 

If you're looking to have the premise of best quality you need to re-train the network and do a few changes in the specifications of the input mel-spectrogram along with the mel-spectrogram used during loss. 

It would be a good idea for input and "loss" mel-spectrograms to match in params. Current configs use an input of 80 bands, fmax 8000 hz but the loss is computed on 80 bands, 11025 hz (default for null is sample-rate / 2). 

The config for "optimal" 22 Khz would need to have as input and output 110 (optional +1) bands, fmax 11025 hz mel-spectrograms. (8000 hz / 80 bands = 100 hz / band => 11025 hz / 100 hz/band = 110.25 bands).

If you want to use the pre-trained hifi-gan - you'll have to retrain for a few steps and change in config the fmax to 11025 khz - it will improve the high-frequency and you should have mel-spec loss of about 0.24-0.25 

Keep in mind that changing the input mel-spectograms means you'll probably have to retrain the synthesizer. If you just change the fmax, you might get away with "fine-tunning" it.

I hope it helps!

Rgs,
Mihai Cvasnievschi
